app/src/main/java/com/asfoundation/wallet/verification/code/VerificationCodeNavigator.kt

```diff
@@ -1,13 +1,21 @@
 package com.asfoundation.wallet.verification.code
 
 import androidx.fragment.app.FragmentManager
+import com.appcoins.wallet.billing.adyen.VerificationCodeResult
+import com.asf.wallet.R
+import com.asfoundation.wallet.verification.VerificationActivityNavigator
 import com.asfoundation.wallet.verification.VerificationActivityView
+import com.asfoundation.wallet.verification.error.VerificationErrorFragment
 
 class VerificationCodeNavigator(private val fragmentManager: FragmentManager,
-                                private val activityView: VerificationActivityView) {
+                                private val activityView: VerificationActivityView,
+                                private val activityNavigator: VerificationActivityNavigator) {
 
   fun navigateToInitialWalletVerification() {
-    fragmentManager.popBackStack()
+    for (i in 0 until fragmentManager.backStackEntryCount) {
+      fragmentManager.popBackStack()
+    }
+    activityNavigator.navigateToWalletVerificationIntro()
```

Comment:
Use this instead
    fragmentManager.popBackStack(null, FragmentManager.POP_BACK_STACK_INCLUSIVE);

Reply:
Same as above.